### PR TITLE
New ARC: ARC-0069: Community Algorand Standard Asset Parameters Conventions for Digital Media Tokens

### PR DESCRIPTION
**Pull request by ChrisAntaki**
This PR adds the popular [ARC69](https://github.com/algokittens/arc69) metadata spec to the `algorandfoundation/ARCs` GitHub repository

This ARC helps NFTs load as quickly and reliably as possible, with attributes, [backwards compatibility for existing NFTs](https://github.com/algokittens/arc69/blob/master/pre-existing-NFTs.md), and updateability (upgrading stats in an RPG, fixing typos, etc.)

The Algo NFT community has enthusiastically adopted this ARC, with more series adding metadata every day! Here are just a few of the awesome series using this ARC:
- Yieldlings
- M.N.G.O
- Eclectic Flamingos
- Banana Mints
- Stupid Horses

image

This ARC works great in production, on popular platforms like [NFT Explorer](https://www.nftexplorer.app/asset/359181888), [ALGOxNFT](https://algoxnft.com/listings/live), and [Rand Gallery](https://www.randgallery.com/algo-collection/?address=359181888)

image
NFT Explorer

Screenshot_20211030-025910
ALGOxNFT

image
Rand Gallery

### Artists
- [Add metadata to your existing NFTs](https://github.com/algokittens/arc69/blob/master/pre-existing-NFTs.md)
### Developers
- [Fetch and parse metadata for NFTs with JavaScript](https://github.com/RandGallery/arc69js)

Thanks for reading this!

### CODE REVIEW

**Comment by ChrisAntaki** [comment]
[ALGOxNFT](https://algoxnft.com/listings/live), the popular auction site for Algo NFTs, now renders ARC69 metadata! 
Screenshot_20211030-025910

**Comment by ChrisAntaki** [comment]
Exciting news! 
image
https://twitter.com/TommasoAE/status/1456748327954001921

**Comment by ChrisAntaki** [comment]
image
https://twitter.com/AndrewWindmills/status/1461204922364358658/

File: ARCs/arc-0069.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+These conventions are heavily based on Ethereum Improvement Proposal [ERC-1155 Metadata URI JSON Schema](https://eips.ethereum.org/EIPS/eip-1155) to facilitate interoperobility. 
+
+The main differences are highlighted below:
+
+* Asset Name, Unit Name, and URL are specified in the ASA parameters. This allows applications to efficienty display meaningful information, even if they aren't aware of ARC-69 metadata.
+* MIME type fields are added to help clients know how to display the files pointed by URI.
+* All asset metadata is stored onchain.
+* Metadata can be either mutable or immutable.
+
+Valid JSON Metadata for ERC-1155 is valid JSON Metadata for ARC-69.
```
Comment (fabrice102) [inline ARCs/arc-0069.md:163]: Is it EIP-721?

File: ARCs/arc-0069.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+Managers MAY update an ASA's ARC69 metadata. To do so, they MUST send a new `acfg` transaction with the entire metadata represented as JSON in the transaction's `note` field.
+
+##### Making ARC69 metadata immutable
+
+Managers MAY make an ASA's ARC69 immutable. To do so, they MUST remove the ASA's manager address with an `acfg` transaction.
+
+## Rationale
+
+These conventions are heavily based on Ethereum Improvement Proposal [ERC-1155 Metadata URI JSON Schema](https://eips.ethereum.org/EIPS/eip-1155) to facilitate interoperobility.
```
Comment (fabrice102) [inline ARCs/arc-0069.md:154]: Is it EIP-721?

File: ARCs/arc-0069.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+```json
+
+{   
+    "standard": "arc69",
+    "description": "arc69 theme song",
+    "external_url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
+    "mime_type":"video/mp4",
+    "attributes": [
+        {"trait_type":"Bass",
```
Comment (fabrice102) [inline ARCs/arc-0069.md:116]: Can you pretty print the JSON?

File: ARCs/arc-0069.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+* *Asset URL*: `ipfs://QmWS1VAdMD353A6SDk9wNyvkT14kyCiZrNDYAad4w1tKqT#v`
+* *Metadata Hash*: the 32 bytes of the SHA-256 digest of the high resolution media file.
+* *Total Number of Units*: 1
+* *Number of Digits after the Decimal Point*: 0
+
+#### Mutability
+
+##### Rendering
+
+Clients SHOULD render an ASA's latest ARC69 metadata. Clients MAY render an ASA's previous ARC69 metadata for changelogs or other historical features.
```
Comment (fabrice102) [inline ARCs/arc-0069.md:142]:
```suggestion
Clients **SHOULD** render an ASA's latest ARC69 metadata. Clients **MAY** render an ASA's previous ARC69 metadata for changelogs or other historical features.

Clients **SHOULD** indicate whether the ARC69 metadata is mutable or not (see below).
```

File: ARCs/arc-0069.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+##### Updating ARC69 metadata
+
+Managers MAY update an ASA's ARC69 metadata. To do so, they MUST send a new `acfg` transaction with the entire metadata represented as JSON in the transaction's `note` field.
+
+##### Making ARC69 metadata immutable
+
+Managers MAY make an ASA's ARC69 immutable. To do so, they MUST remove the ASA's manager address with an `acfg` transaction.
```
Comment (fabrice102) [inline ARCs/arc-0069.md:150]:
```suggestion
Managers **MAY** make an ASA's ARC69 immutable. To do so, they **MUST** remove the ASA's manager address with an `acfg` transaction.

```

File: ARCs/arc-0069.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+#### Mutability
+
+##### Rendering
+
+Clients SHOULD render an ASA's latest ARC69 metadata. Clients MAY render an ASA's previous ARC69 metadata for changelogs or other historical features.
+
+##### Updating ARC69 metadata
+
+Managers MAY update an ASA's ARC69 metadata. To do so, they MUST send a new `acfg` transaction with the entire metadata represented as JSON in the transaction's `note` field.
```
Comment (fabrice102) [inline ARCs/arc-0069.md:146]:
```suggestion
Managers **MAY** update an ASA's ARC69 metadata. To do so, they **MUST** send a new `acfg` transaction with the entire metadata represented as JSON in the transaction's `note` field.

```

File: ARCs/arc-0069.md (hunk 5 above)
Comment (fabrice102) [reply ARCs/arc-0069.md:150]: In that case, should managers repeat the note field with metadata or can they leave the note field empty?

File: ARCs/arc-0069.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
```
Comment (fabrice102) [inline ARCs/arc-0069.md:147]:
```suggestion

Pre-ARC69 NFTs can be updated to ARC69 in a similar way as long as the manager address is present and the ASA creator has control over it.
```

File: ARCs/arc-0069.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    * **SHOULD NOT** use the following URI scheme: *http* (due to security concerns).
+* *Asset Metadata Hash* (`am`): the SHA-256 digest of the full resolution media file as a 32-byte string (as defined in [NIST FIPS 180-4](https://doi.org/10.6028/NIST.FIPS.180-4))
+    * **OPTIONAL**
+* *Freeze Address* (`f`): 
+    * **SHOULD** be empty, unless needed for royalties or other use cases
+* *Clawback Address* (`c`): 
+    * **SHOULD** be empty, unless needed for royalties or other use cases
+
+
+There are no requirements regarding the manager account of the ASA, or the reserve account. However, if immutability is required the manager address **MUST** be removed.
```
Comment (fabrice102) [inline ARCs/arc-0069.md:48]:
```suggestion
There are no requirements regarding the manager account of the ASA, or the reserve account. However, if immutability is required the manager address **MUST** be removed.

Furthermore, the manager address, if present, **SHOULD** be under the control of the ASA creator, as the manager address can unilaterally change the metadata. Some advanced use cases **MAY** use a logicsig as ASA manager, if the logicsig only allows to set the note fields by the ASA creator.
```

**Comment by fabrice102** [comment]
@ChrisAntaki @ipaleka and others, thank you for your contribution.

Following the above discussion and adoption, we plan to merge this as an official ARC first in "draft" and very soon in "final call" setting.

Because of backward compatibility, we exceptionally will keep the ARC-0069 numbering.
Future ARCs will follow ARC-0000 which will most likely require incremental numbers for clarity.

I see a few additional points of discussion and would be happy to have the point of view of @ChrisAntaki and the community in general. All the comments below ensure complete backward compatibility with the current ARC69 standard and try to keep the spirit of the standard.

## Mime Types

@TommyAlex79 asked whether other mime types may be supported for the asset pointed by the asset URL.
Would it make sense to allow the fragment `#o` for others, and then specify the mime type in metadata?

## Restrictions on asset names and URLs

I agree with @ChrisAntaki that there is no need to specify the protocol restrictions there.
However, should we consider to allow specifying a longer name in metadata in case the asset name is too short, like @ipaleka proposed?

## Media_url vs Asset URL

Is `media_url` mandatory? 
If it is not, is the metadata hash the SHA256 hash of the digital media file in asset URL?
Also, is the mimetype of `media_url` the same as the asset URL?

## EIP-1155 vs EIP-721 metadata

EIP-1155 uses "properties" instead of "attributes".

Would you consider allowing either "properties" or "attributes"?

## Integrity

I am strongly in favor of providing strong integrity mechanisms, as in ARC-0003.
Without those, if the medias are no more available and an S3 bucket was used, there is no way to recover it.
Would it be possible to add the same `_integrity` fields as ARC-0003?

This is completely backward compatible and clients may ignore it.

## Compatibility with ARC-0002

ARC-0002 (https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0002.md) recommend the use of a prefix `arc69:j` to the metadata field.
Would it be reasonable to allow this prefix as an option? (not mandatory for backward compatibility but recommended)

## Enforcing CORS

Should we enforce CORS headers as ARC-0003 does:

> MUST be such that the returned resource includes the CORS header
> Access-Control-Allow-Origin: *
> if the URI scheme is https
> This requirement is to ensure that client JavaScript can load all resources pointed by https URIs inside an ARC-3 ASA.

This is not as important as for ARC-0003 in that usually the media type would be displayed directly by the browser. But I could see cases where it is not, e.g., if it needs to be interpreted to be displayed.

## Comments as review

I've also made a few comments in the review.

File: ARCs/arc-0069.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+
+While mutability helps with backwards compatibility and other use cases, like leveling up an RPG character, some use cases call for immutability. In these cases, the ASA manager MAY remove the manager address, after which point the Algorand network won't allow anyone to send asset configuration transactions for the ASA. This effectively makes the latest valid ARC-69 metadata immutable.
+
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+An ARC-69 ASA has an associated JSON Metadata file, formatted as specified below, that is stored on-chain in the most recent asset configuration transaction.
```
[the remark this answers is not in the document]
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:24]: We could make this even more accurate. It's the most recent `acfg` with valid ARC69 metadata in the `note` field

File: ARCs/arc-0069.md (hunk 8 above)
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:48]: Nice

File: ARCs/arc-0069.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+    * **SHOULD** be empty, unless needed for royalties or other use cases
+* *Clawback Address* (`c`): 
+    * **SHOULD** be empty, unless needed for royalties or other use cases
+
+
+There are no requirements regarding the manager account of the ASA, or the reserve account. However, if immutability is required the manager address **MUST** be removed.
+
+### JSON Metadata File Schema
+
+> The JSON Medata File schema follow the Ethereum Request for Comments 721 [ERC-721 Metadata URI JSON Schema](https://eips.ethereum.org/EIPS/eip-1155) with the following main differences:
```
[the remark this answers is not in the document]
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:52]: Might be better to just remove the reference to either EIP here

File: ARCs/arc-0069.md (hunk 4 above)
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:142]: > Clients **SHOULD** indicate whether the ARC69 metadata is mutable or not (see below).

I'd prefer to leave this up to clients. Currently none of them indicate whether ARC69 metadata is mutable, and that seems like a product decision

File: ARCs/arc-0069.md (hunk 7 above)
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:148]: I think this is a little redundant. The previous paragraph says "Managers MAY update an ASA", which implies (1) a manager address is present and (2) all ASAs are supported

File: ARCs/arc-0069.md (hunk 5 above)
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:151]: They can leave the `note` field empty 👍

File: ARCs/arc-0069.md (hunk 1 above)
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:165]: I suppose the only field we share with `EIP-721` is `description`. Removing this section is probably best

File: ARCs/arc-0069.md (hunk 2 above)
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:156]: I suppose the only field we share with `EIP-721` is `description`. Removing this section is probably best

File: ARCs/arc-0069.md (hunk 3 above)
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:116]: Definitely

**Comment by ChrisAntaki** [comment]
> Following the above discussion and adoption, we plan to merge this as an official ARC first in "draft" and very soon in "final call" setting.

That's very exciting

> Because of backward compatibility, we exceptionally will keep the ARC-0069 numbering.
Future ARCs will follow ARC-0000 which will most likely require incremental numbers for clarity.

I appreciate you making an exception in this case

> @TommyAlex79 asked whether other mime types may be supported for the asset pointed by the asset URL.
Would it make sense to allow the fragment #o for others, and then specify the mime type in metadata?

@TommyAlex79 and I discussed this more, and we like:
- The `#o` idea, which tells clients the ARC69 metadata contains a useful MIME type
- @runvnc 's idea to allow full MIME types in the URL fragment for efficiency, space permitting (https://github.com/algokittens/arc69/issues/11)
- Several new shorthand fragments to cover the remaining discrete types, except for "application" since it's so wide-ranging (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#discrete_types)
  - `#f` for font
  - `#m` for model
  - `#t` for text

> EIP-1155 uses "properties" instead of "attributes".
Would you consider allowing either "properties" or "attributes"?

I prefer staying simple and sticking with Open Sea's metadata standards (https://docs.opensea.io/docs/metadata-standards#metadata-structure), which call for an `attributes` array

>I am strongly in favor of providing strong integrity mechanisms, as in ARC-0003.
Without those, if the medias are no more available and an S3 bucket was used, there is no way to recover it.
Would it be possible to add the same _integrity fields as ARC-0003?
This is completely backward compatible and clients may ignore it.

That's a good point, it definitely is a factor with non-IPFS URLs. Adding these fields as optional **MAY**s, with guidance for when they make sense, seems smart

>ARC-0002 (https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0002.md) recommend the use of a prefix arc69:j to the metadata field.
Would it be reasonable to allow this prefix as an option? (not mandatory for backward compatibility but recommended)

I prefer to keep it simple and keep the JSON value. Adding that prefix would of course make the `note` invalid JSON, which would break existing parsing code and require more complexity on the client

> Should we enforce CORS headers as ARC-0003 does

Yeah, adding this suggestion as a **SHOULD**, with guidance for when it makes sense, seems smart

File: ARCs/arc-0069.md [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+> * Adding the field `standard` to signal which standard is used.
+> * Adding the field `external_url` used by [OpenSea metadata format](https://docs.opensea.io/docs/metadata-standards).
+> * Adding the field `mime_type` to signal the MIME type and subtype of the media included in `au`. **OPTIONAL**
```
[the remark this answers is not in the document]
Comment (ChrisAntaki) [reply ARCs/arc-0069.md:55]: Regarding the `mime_type` field, the spec refer to the ASA URL when it says: "Adding the field `mime_type` to signal the MIME type and subtype of the media included in `au`. **OPTIONAL**"

Adding a `media_mime_type` field seems smart

**Comment by ChrisAntaki** [comment]
> I agree with @ChrisAntaki that there is no need to specify the protocol restrictions there.
However, should we consider to allow specifying a longer name in metadata in case the asset name is too short, like @ipaleka proposed?

Using the Asset Name has been working great. It's available ASAP, and it naturally limits names to a reasonable length

File: ARCs/arc-0069.md (hunk 7 above)
Comment (wheresaddie) [reply ARCs/arc-0069.md:148]: I would either then add this statement to the prior paragraph or include it as is.

Comment (ChrisAntaki) [reply ARCs/arc-0069.md:148]: Good call! What if this paragraph just replaced the paragraph on line 146?

> If an ASA has a manager address, then the manager **MAY** update the ASA's ARC69 metadata. To do this, the manager sends an `acfg` transaction with a `note` field containing ARC69 metadata.

**Comment by ChrisAntaki** [comment]
Hey, just curious what my next steps are? Were there any "dealbreakers" for the foundation in my responses, or would it make sense for me to go forward with changes to the PR?

**Comment by ChrisAntaki** [comment]
Please let me know if/when the foundation feels like engaging on this PR, since it lives in their centralized repository. Until then, I can propose changes in ARC69's official repository https://github.com/algokittens/arc69 ☯️

**Comment by barnjamin** [comment]
I think the next step is to PR whatever changes you have then Fabrice/Addie can merge

**Comment by ChrisAntaki** [comment]
Cool, so I see both you and @wheresaddie gave my last sizeable comment (https://github.com/algorandfoundation/ARCs/pull/38#issuecomment-991392305) a 👍 reaction. That's encouraging, especially if @wheresaddie can approve and merge this PR eventually!

I notice @fabrice102 hasn't responded though. When he left his last sizeable comment (https://github.com/algorandfoundation/ARCs/pull/38#issuecomment-991341809) on Dec 10th, that was the first time Algo Inc/Foundation responded to this PR since October 29th. That, plus his leadership of the ARC3 design, gave me the impression @fabrice102 was leading the review of this PR too

I'm just trying to understand the foundation's PR review protocol here. If @fabrice102 stops responding to comments or changes, can the PR still get approved and merged?

**Comment by fabrice102** [comment]
I think we agree on almost everything. @ChrisAntaki 

Sorry for the delay, yes please make the above changes we discussed:
* adding mimetype
* adding integrity
* enforcing CORS
* the minor style change mentioned above (you can easily commit those by click on a button)

If you did not see in practice the need for longer names, I guess we can leave this open.
In the worse case, we can have an extension of ARC-69 if this is blocking in the future.

The two points we can still discuss are:

> > EIP-1155 uses "properties" instead of "attributes".
> > Would you consider allowing either "properties" or "attributes"?
> 
> I prefer staying simple and sticking with Open Sea's metadata standards (https://docs.opensea.io/docs/metadata-standards#metadata-structure), which call for an `attributes` array. Also, changing the field name to `properties` would break existing parsing code and supporting both would require more complexity on the client

Actually, OpenSea explicitly allows for `properties` with the EIP-1155 format in addition to `attributes`:
https://docs.opensea.io/docs/metadata-standards#metadata-structure

I understand it makes parsing slightly more difficult, but I think it would be great.
This increases compatibility with existing EIP-1155 assets.

What do you think?

> > ARC-0002 (https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0002.md) recommend the use of a prefix arc69:j to the metadata field.
> > Would it be reasonable to allow this prefix as an option? (not mandatory for backward compatibility but recommended)
> 
> I prefer to keep it simple and keep the JSON valid. Adding that prefix would of course make the `note` invalid JSON, which would break existing parsing code and require more complexity on the client

I think this can be checked and fixed like this:
```ts
const arc69note = "...";
let arc69json = arc69note;
if(arc69json.startsWith("arc69:j")) {
  arc69json = arc69json.substring(7);
}
```

What do you think?

**Comment by ChrisAntaki** [comment]
Thanks for your response, Fabrice

So a week ago, you said "All the comments below ensure complete backward compatibility with the current ARC69 standard". Some of your comments broke backwards compatibility though. Two of those cases were the EIP-1155 `properties` object, and the `arc69:j` note prefix

In both cases, I'd prefer backwards compatibility

**Comment by ChrisAntaki** [comment]
Looks like adding support for EIP-1155 `properties` would add a _lot_ of complexity too. For example, they allow people to specify arbitrary CSS
image

**Comment by ChrisAntaki** [comment]
What do you think, should I proceed with this PR if I don't add the note prefix or the `properties` object?

**Comment by TommyAlex79** [comment]
I see "EIP-1155 properties" as a candidate for ARC3 extension or some other more comprehensive ARC. ARC69 is meant to be simple and pragmatic. Inevitably, a more complex standard, and without size limitation, will be required to address most use-cases, and not just some. 

Regarding "arc69:j", this would break backward compatibility (50K plus arc69 assets, and all dapps/apps already reading arc69 too, of course). In principle I have no objection, if we were starting from scratch.

**Comment by ChrisAntaki** [comment]
After some more discussion, the "simple" properties from EIP-1155 seem like a good fit, since they're more space efficient and readable

File: ARCs/arc-0069.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (fabrice102) [reply ARCs/arc-0069.md:24]: valid ARC69 metadata may be subject to interpretation.
What about with a valid JSON in the `note` field?

File: ARCs/arc-0069.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (fabrice102) [reply ARCs/arc-0069.md]: ok

**Comment by fabrice102** [comment]
Just a heads up that I won't be available until new year.
But I don't see any road-blocker for finalization of this ARC by mid-January.

We won't add the `arc69:j` prefix.

@ChrisAntaki Feel free to add the simple properties from EIP-1155.
There are a few other comments I'll do a PR about if needed (`_integrity` and `_mime_type`).
None of this requires any update from any client nor any ASA.

**Comment by fabrice102** [comment]
I'm sorry I completely dropped the ball there. I don't see any comments so I'll merge it with the above changes `_integrity` and `_mime_type` tomorrow, as a last call and then finalized.

**Comment by ChrisAntaki** [comment]
Hey Fabrice! So the PR currently proposes a status of "Living Standard", inspired by the HTML spec, which allows for upgrades over time

**Comment by ChrisAntaki** [comment]
Also could you create suggestions for the `_integrity` and `_mime_type` fields in this PR, so I could review it ?

**Comment by ChrisAntaki** [comment]
Sidenote: 52.6% of Algo NFTs use ARC69 now ☯️
https://www.nftexplorer.app/stats

File: ARCs/arc-0069.md (hunk 8 above)
Comment (fabrice102) [reply ARCs/arc-0069.md:48]: merging manually

File: ARCs/arc-0069.md (hunk 4 above)
Comment (fabrice102) [reply ARCs/arc-0069.md:135]: ok, let's not add this

File: ARCs/arc-0069.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (fabrice102) [reply ARCs/arc-0069.md:24]: merging manually

**Comment by fabrice102** [comment]
Usually we expect ARCs to be finalized so they don't change.
Then if changes occur, we create additional ARCs to add/modify behaviors.
That's definitely what we should aim at in the future.

But since the process was not fully in place (and still is not fully in place) when ARC-69 was in place, let's start with merging it as a living standard, as an exception.

I cannot push on your PR, so:
* I've manually merged your PR with your three approved comments (see changes https://github.com/algorandfoundation/ARCs/commit/4bacf16be09b02378a3823a1c6dc5785de7c9f9c)
* I've open a new issue to discuss ARC-69: 
* I've opened a new PR for `_integrity` and `_mime_type`: https://github.com/algorandfoundation/ARCs/pull/65

**Comment by ChrisAntaki** [comment]
> let's start with merging it as a living standard, as an exception

Sounds good!

**Pull request merged**